firedrake/functionspaceimpl.py

```diff
@@ -403,6 +403,13 @@ def make_function_space(cls, mesh, element, name=None):
             new = cls.create(new, mesh)
         return new
 
+    def broken_space(self):
+        """Return a :class:`.WithGeometryBase` with a BrokenElement
+        constructed from this function space's FiniteElement."""
+        return type(self).make_function_space(
+            self.mesh(), finat.ufl.BrokenElement(self.ufl_element()),
+            name=f"{self.name}_broken")
```

Comment:
Can this result in `"None_broken"`? Maybe don't set a `name` if `self.name is None`?